app/src/main/cpp/skyline/kernel/types/KPrivateMemory.cpp

```diff
@@ -8,66 +8,63 @@
-        if (mprotect(ptr, nSize, PROT_READ | PROT_WRITE | PROT_EXEC) < 0)
+        if (mprotect(guest.data(), nSize, PROT_READ | PROT_WRITE | PROT_EXEC) < 0)
             throw exception("An occurred while resizing private memory: {}", strerror(errno));
 
-        if (nSize < size) {
+        if (nSize < guest.size()) {
             state.process->memory.InsertChunk(ChunkDescriptor{
-                .ptr = ptr + nSize,
-                .size = size - nSize,
+                .ptr = guest.data() + nSize,
+                .size = guest.size() - nSize,
                 .state = memory::states::Unmapped,
             });
-        } else if (size < nSize) {
+        } else if (guest.size() < nSize) {
             state.process->memory.InsertChunk(ChunkDescriptor{
-                .ptr = ptr + size,
-                .size = nSize - size,
+                .ptr = guest.data() + guest.size(),
+                .size = nSize - guest.size(),
                 .permission = permission,
                 .state = memoryState,
             });
         }
-
-        size = nSize;
+        guest = span<u8>(guest.data(), nSize);
     }
 
-    void KPrivateMemory::Remap(u8 *nPtr, size_t nSize) {
-        if (!state.process->memory.base.IsInside(nPtr) || !state.process->memory.base.IsInside(nPtr + nSize))
-            throw exception("KPrivateMemory remapping isn't inside guest address space: 0x{:X} - 0x{:X}", nPtr, nPtr + nSize);
-        if (!util::IsPageAligned(nPtr) || !util::IsPageAligned(nSize))
-            throw exception("KPrivateMemory remapping isn't page-aligned: 0x{:X} - 0x{:X} (0x{:X})", nPtr, nPtr + nSize, nSize);
+    void KPrivateMemory::Remap(span<u8> map) {
+        if (!state.process->memory.base.contains(map))
+            throw exception("KPrivateMemory remapping isn't inside guest address space: 0x{:X} - 0x{:X}", map.data(), map.end().base());
+        if (!util::IsPageAligned(map.data()) || !util::IsPageAligned(map.size()))
+            throw exception("KPrivateMemory remapping isn't page-aligned: 0x{:X} - 0x{:X} (0x{:X})",  map.data(), map.end().base(), map.size());
 
-        if (mprotect(ptr, size, PROT_NONE) < 0)
+        if (mprotect(guest.data(), guest.size(), PROT_NONE) < 0)
             throw exception("An occurred while remapping private memory: {}", strerror(errno));
 
-        if (mprotect(nPtr, nSize, PROT_READ | PROT_WRITE | PROT_EXEC) < 0)
+        if (mprotect(map.data(), map.size(), PROT_READ | PROT_WRITE | PROT_EXEC) < 0)
             throw exception("An occurred while remapping private memory: {}", strerror(errno));
     }
 
-    void KPrivateMemory::UpdatePermission(u8 *pPtr, size_t pSize, memory::Permission pPermission) {
-        pPtr = std::clamp(pPtr, ptr, ptr + size);
-        pSize = std::min(pSize, static_cast<size_t>((ptr + size) - pPtr));
+    void KPrivateMemory::UpdatePermission(span<u8> map, memory::Permission pPermission) {
+        auto pPtr = std::clamp(map.data(), guest.data(), guest.end().base());
```

Comment:
drop p and use {}, p is for parameters and it's no longer from a parameter henc enot needed